Dockerfile.tmpl

```
@@ -255,196 +260,193 @@ RUN apt-get install -y libgl1-mesa-glx && \
-    pip install fury dipy && \
-    pip install plotnine && \
-    pip install scikit-surprise && \
-    pip install pymongo && \
-    pip install geoplot && \
-    pip install eli5 && \
-    pip install kaggle && \
-    pip install mock && \
+    pip install deap \
+        tpot \
+        scikit-optimize \
+        haversine \
+        toolz cytoolz \
+        plotly \
+        hyperopt \
+        fitter \
+        langid \
+        # Delorean. Useful for dealing with datetime
+        delorean \
+        trueskill \
+        # Useful data exploration libraries (for missing data and generating reports)
+        missingno \
+        pandas-profiling \
+        s2sphere \
+        bayesian-optimization \
+        matplotlib-venn \
+        # b/184083722 pyldavis >= 3.3 requires numpy >= 1.20.0 but TensorFlow 2.4.1 / 2.5.0 requires 1.19.2
+        pyldavis==3.2.2 \
+        mlxtend \
+        altair \
+        ImageHash \
+        ecos \
+        CVXcanon \
+        pymc3 \
+        imagecodecs \
+        tifffile \
+        spectral \
+        descartes \
+        geojson \
+        pydicom \
+        wavio \
+        SimpleITK \
+        hmmlearn \
+        bayespy \
+        gplearn \
+        PyAstronomy \
+        squarify \
+        fuzzywuzzy \
+        python-louvain \
+        pyexcel-ods \
+        sklearn-pandas \
+        stemming \
+        # b/266272046 prophet 1.1.2 breaks the test
+        prophet==1.1.1 \
+        holoviews \
+        geoviews \
+        hypertools \
+        py_stringsimjoin \
+        mlens \
+        scikit-multilearn \
+        cleverhans \
+        leven \
+        catboost \
+        lightfm \
+        folium \
+        scikit-plot \
+        fury dipy \
+        plotnine \
+        scikit-surprise \
+        pymongo \
+        geoplot \
+        eli5 \
+        kaggle \
+        mock && \
     /tmp/clean-layer.sh
 
 RUN pip install tensorpack && \
     # Add google PAIR-code Facets
     cd /opt/ && git clone https://github.com/PAIR-code/facets && cd facets/ && jupyter nbextension install facets-dist/ --user && \
     export PYTHONPATH=$PYTHONPATH:/opt/facets/facets_overview/python/ && \
-    pip install pycountry && \
-    pip install iso3166 && \
-    pip install pydash && \
     pip install kmodes --no-dependencies && \
-    pip install librosa && \
-    pip install polyglot && \
-    pip install mmh3 && \
-    pip install fbpca && \
-    pip install sentencepiece && \
-    pip install cufflinks && \
-    pip install lime && \
-    pip install memory_profiler && \
+    pip install librosa \
+        polyglot \
+        mmh3 \
+        fbpca \
+        sentencepiece \
+        cufflinks \
+        lime \
+        memory_profiler && \
     /tmp/clean-layer.sh
 
 # install cython & cysignals before pyfasttext
-RUN pip install --upgrade cython && \
-    pip install --upgrade cysignals && \
-    pip install pyfasttext && \
-    pip install fasttext && \
+RUN pip install cython && \
```

Whoops, I missed that one in the last pass. Same issue.

---

what do you mean by miss? the `-U` was dropped as if you install on a clean system it uses the latest version by default, and if you upgrade it this late in the docker building process, you may break previously installed packages...

---

@Borda I meant the && \ is on this line like it was for the other one, which makes the following python packages be interpreted as bash commands instead of a list of packages to pip.

---

```suggestion
RUN pip install cython \
```
yep see it now :innocent: